### PR TITLE
Fix for HomeKit fan modes (issue #563)

### PR DESCRIPTION
**Pull request by diegocjorge**
All credits to @kparmun for the fix:

https://github.com/ollo69/ha-smartthinq-sensors/issues/563#issuecomment-1612297396

### CODE REVIEW

**Comment by ollo69** [comment]
It doesn't look really good alter an enum using in some cases lower case and in other upper case. If the problem is that HomeKit just support lower case options, the change should be implemented in the methods `fan_speeds`, `fan_speed` and `set_fan_speed` to convert the values in lower case.

**Comment by diegocjorge** [comment]
Hey @ollo69, hope it's better now. I'm not much of a coder.

File: custom_components/smartthinq_sensors/wideq/devices/ac.py [hunk 1]
```diff
@@ -690,6 +690,8 @@ async def set_fan_speed(self, speed):
         """Set the fan speed to a value from the `ACFanSpeed` enum."""
         if speed not in self.fan_speeds:
             raise ValueError(f"Invalid fan speed: {speed}")
+        # Necessary for HomeKit compatibility
+        speed = speed.upper()
         keys = self._get_cmd_keys(CMD_STATE_WIND_STRENGTH)
         speed_value = self.model_info.enum_value(keys[2], ACFanSpeed[speed].value)
```
Comment (ollo69) [inline custom_components/smartthinq_sensors/wideq/devices/ac.py:696]:
```suggestion
        keys = self._get_cmd_keys(CMD_STATE_WIND_STRENGTH)
        # Convert upper() necessary for HomeKit compatibility
        speed_value = self.model_info.enum_value(keys[2], ACFanSpeed[speed.upper()].value)
```

**Review by ollo69 — CHANGES_REQUESTED** [review]
Please take a look to suggested changes and fix black formatting

File: custom_components/smartthinq_sensors/wideq/devices/ac.py [hunk 2]
```diff
@@ -690,6 +690,8 @@ async def set_fan_speed(self, speed):
         """Set the fan speed to a value from the `ACFanSpeed` enum."""
         if speed not in self.fan_speeds:
```
Comment (ollo69) [inline custom_components/smartthinq_sensors/wideq/devices/ac.py:691]:
```suggestion
        if speed.lower() not in self.fan_speeds:
```

Comment (ollo69) [reply custom_components/smartthinq_sensors/wideq/devices/ac.py:691]: Probably is better to also put here a lower() to not break existing configured service calls

**Comment by diegocjorge** [comment]
I committed the changes you recommended

**Comment by ollo69** [comment]
Not sure this is the right solution. Probably fan modes should be mapped with [HA modes](https://developers.home-assistant.io/docs/core/entity/climate#fan-modes).
Let me think about this.

**Review by ollo69 — APPROVED** [review]

**Pull request merged**